minicriu.c

```diff
@@ -183,12 +183,12 @@ int main(int argc, char *argv[]) {
 		if (ph->p_type != PT_LOAD) {
 			continue;
 		}
-		if (munmap((void*)ph->p_vaddr, ph->p_filesz)) {
+		if (munmap((void*)ph->p_vaddr, ph->p_memsz)) {
 			/*perror("munmap");*/
 		}
 		void *addr = mmap((void*)ph->p_vaddr,
-				ph->p_filesz,
-				PROT_READ | PROT_WRITE | PROT_EXEC,
+				ph->p_memsz,
+				PROT_WRITE,
```

Comment:
There is no need to minimize permissions here, they will be fixed later anyway. Let's add PROT_READ at least to avoid unnecessary problems, mmap(2):
```
       On some hardware architectures (e.g., i386), PROT_WRITE implies PROT_READ.  It is architecture dependent whether PROT_READ implies PROT_EXEC or not.  Portable programs should always set PROT_EXEC if
       they intend to execute code in the new mapping.
```